### PR TITLE
[C#] Add PerfTest

### PR DESCRIPTION
**Pull request by TedHartMS**

### CODE REVIEW

File: cs/PerfTest/ParseArgs.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                }
+                Console.WriteLine($"Usage: Specify parameters for one or more test runs, from either a {TestFileArg} or from command-line arguments.");
+                Console.WriteLine("       Parameter values default to the fastest configuration (use read cache, flush-only log, no object values).");
+                Console.WriteLine();
+                Console.WriteLine("  Test parameters from a file:");
+                Console.WriteLine($"    {TestFileArg} <fname>: The name of a JSON file containing {nameof(TestParameters)}, such as in the 'testfiles' subdirectory.");
+                Console.WriteLine();
+                Console.WriteLine($"  Test parameters from individual command-line arguments (overrides those specified in {TestFileArg} (if any)):");
+                Console.WriteLine($"    {UpsertsArg} <num>: The number of Upserts for keys in 0..({UpsertsArg})-1; default is {defaultTestResult.UpsertCount}");
+                Console.WriteLine($"    {DataArg} <bytes>: How many bytes of per CacheValue; default {CacheGlobals.MinDataSize} (minimum; sizeof(long))");
```
Comment (badrishc) [inline cs/PerfTest/ParseArgs.cs]: Is the key size configurable as well?

File: cs/PerfTest/ParseArgs.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                Console.WriteLine("             Useful for comparing test-parameter change impact.");
+                Console.WriteLine();
+                Console.WriteLine("  Common parameters:");
+                Console.WriteLine($"    {ResultsFileArg} fname: The name of file to receive JSON text containing either:");
+                Console.WriteLine($"        {nameof(TestResultComparisons)}, if {CompareResultsExactArg} or {CompareResultsSequenceArg} was specified");
+                Console.WriteLine($"        {nameof(TestResults)} otherwise");
+                Console.WriteLine($"    {VerboseArg}: Verbose; print status messages; default = {verbose}");
+                Console.WriteLine($"    {PromptArg}: Print a message and wait before exiting; default = {prompt}");
+                Console.WriteLine();
+                return false;
```
Comment (badrishc) [inline cs/PerfTest/ParseArgs.cs]: LogSettings has various settings that are configurable for the log, e.g., mutable fraction, total memory pages, segment size, etc. How do we control these in runs?

File: cs/PerfTest/PerfTest.cs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+            {
+                Console.WriteLine("No tests were run");
+                return;
+            }
+
+            Console.WriteLine($"Completed {results.Results.Length} test run(s) in {TimeSpan.FromMilliseconds(sw.ElapsedMilliseconds)}");
+            if (!string.IsNullOrEmpty(resultsFilename))
+            {
+                results.Write(resultsFilename);
+                Console.WriteLine($"Results written to {resultsFilename}");
```
Comment (badrishc) [inline cs/PerfTest/PerfTest.cs]: How does performance compare to FASTER.benchmark when settings are identical? So we have a reference for the performance behavior.

File: cs/PerfTest/PerfTest.cs [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+            fht.Close();
+
+            if (prompt)
+            {
+                Console.WriteLine("Press <ENTER> to end");
+                Console.ReadLine();
+            }
+        }
+
+        private static void RunUpserts<TValue, TOutput, TFunctions, TSerializer>(
```
Comment (badrishc) [inline cs/PerfTest/PerfTest.cs]: Run this multi-threaded (configurable) as well?

**Comment by badrishc** [comment]
Linux tests are failing in this PR, probably because of the project name changes.

**Comment by badrishc** [comment]
Could you bring this PR up to date? It would be good to merge this soon. Thanks!

**Comment by TedHartMS** [comment]
Closed for now due to other priorities

**Pull request closed without merging**